README.md

```diff
@@ -1,29 +1,64 @@
 # check_multiple
 
-Nagios plugin used to execute multiple commands in parlallel. It can be configured to return failure if one command fails, or only if they all fail.
+Nagios plugin used to execute multiple commands in parallel.
+
+With the `--mode` parameter, it can be configured to return either the
+best overall result, or the worst. If you choose the worst overall
+result (the default), then you will be notified when one of the
+commands fails. If you choose the best overall result, then you will
+be notified only when all of them fail.
+
+The output from check_multiple indicates how many OK, WARNING, and
+CRITICAL results the individual commands produced. Afterwards, the
+output from those individual commands, concatenated onto one line per
+command, is emitted.
+
+An example Nagios command that will ping a host and check that
+HTTP/SSH are running on it, and return the best overall result
+(i.e. the host is up if any service responds):
 
-Example command that will check SSH and ping host:
 ```
-define command{
-command_name check-host-alive-multiple
+define command {
+  command_name check-host-alive-multiple
 
-command_line $USER1$/check_multiple --mode one --commands "$USER1$/check_ping -H $HOSTADDRESS$ -w 1000.0,80% -c 1000.0,100% -p 5@#%$USER1$/check_http -H $HOSTADDRESS$@#%$USER1$/check_ssh -H $HOSTADDRESS$"
+  command_line $USER1$/check_multiple --mode best "$USER1$/check_ping -H $HOSTADDRESS$ -w 1000.0,80% -c 1000.0,100% -p 5" "$USER1$/check_http -H $HOSTADDRESS$" "USER1$/check_ssh -H $HOSTADDRESS$"
 }
 ```
 
-Run from the command line:
+Command-line usage:
+
 ```
-usage: check_multiple.py [-h] [--mode {all,one}] commands [commands ...]
+usage: check_multiple [-h] [--mode {worst,best}] command [command ...]
 
-Run multiple Nagios checks at once.
+Run multiple Nagios checks and combine the results.
 
 positional arguments:
-  commands          Checks to run. Enclose (separately) in quotes.
+  command              check to run, enclosed in quotes
 
 optional arguments:
-  -h, --help        show this help message and exit
-  --mode {all,one}  Specify `one` or `all`. `one` will return OK if any one of
-                    the checks is successfull. `all` will return CRITICAL if
-                    any one of the checks fail. default is `all`.
+  -h, --help           show this help message and exit
+  --mode {worst,best}  which individual check result should be the overall
```

Comment:
i never was satisfied with "all/one". it is not clear what it means, and there is no precedent for the pattern. is there a reason you have used "best/worst"? maybe "all/any", which would be analogous with python's all() and any()?

another thought is to change the param to a bool, "optimistic". if false (default), return success if all pass. if true: return success if any pass). the downside to this solution is that we would have to change back to a string if we every want to add more than two "modes", whereas above we could add "none", "one" down the road if needed...

in any case this is fine, just doing some bikeshedding.

Reply:
"Any" and "all" no longer match up nicely to the possible outputs. I added the ability to report warnings from the individual checks, and to return WARNING as a result. The problem I then ran into was that the names "all" or "one" don't let you specify what to do when you see WARNING.

With "best," for example, you can say for example that if you see CRITICAL/WARNING, then you want to report WARNING. Likewise with "worst" you can return WARNING from WARNING/OK. This is "backwards compatible" in the sense that it still matches up with the previous behavior if all you ever see is CRITICAL and/or OK, but the names are now more descriptive when WARNING is involved. For the same reason, "any" and "all" are confusing in that context. Quick, what should "any" return when you see CRITICAL/WARNING? I have to stop and think; but with best/worst it's immediately clear.

As far as I can see, an "optimistic" flag would work though. I just worry that the documentation for that option is going to involve the words "best" and "worst" =)

Reply:
ok good enough for me.